WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -0,0 +1,206 @@
+import Foundation
+import UIKit
+
+public typealias ImmuTableActionType = (ImmuTableRow) -> Void
+
+public protocol Reusable {
+    static var reusableIdentifier: String { get }
+}
+
+extension Reusable {
+    var reusableIdentifier: String {
+        get {
+            return self.dynamicType.reusableIdentifier
+        }
+    }
+}
+
+public protocol ImmuTableRow: Reusable {
+    var action: ImmuTableActionType? { get }
+    static var cellClass: AnyClass { get }
+    func configureCell(cell: UITableViewCell)
+    static var registrable: CellRegistrable { get }
+    static var customHeight: Float? { get }
+}
+
+public struct ImmuTableSection {
+    let headerText: String?
+    let rows: [ImmuTableRow]
+    let footerText: String?
+
+    init(rows: [ImmuTableRow]) {
+        self.headerText = nil
+        self.rows = rows
+        self.footerText = nil
```

Comment:
Optionals don't default to nil, and still need to be initialised

![screen shot 2015-12-08 at 14 28 29](https://cloud.githubusercontent.com/assets/8739/11656844/00a562a6-9db8-11e5-8f19-dd659b239b89.png)